### PR TITLE
Run pull requests on github actions

### PR DESCRIPTION
**Pull request by e00E**
I'm not yet sure what the best way to do deployments is so they continue
to be run by travis.
ganache is run without docker because while github actions supports
docker services I could not figure out how to pass arguments to them.

### Test Plan
Github actions should run for this PR.
After merging travis should run for the deployment.
Also check that merge when green works with this PR.

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  rust:
+    runs-on: ubuntu-latest
+    env:
+      CARGO_INCREMENTAL: 0
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          npm install ganache-cli
+          node_modules/.bin/ganache-cli --networkId 5777 --gasLimit 10000000 --gasPrice 0&
```
Comment (fleupold) [inline .github/workflows/pull-request.yaml:12]: nit: for debugging purposes, is it possible to pipe the output into a file and cat it at the end?

File: .travis.yml [hunk 2]
```diff
@@ -7,44 +7,9 @@ os: linux
 dist: focal
 
 jobs:
-  fast_finish: true
   include:
-    - name: "Test Rust"
-      language: rust
-      rust: stable
-      env:
-        - SCCACHE_VERSION=0.2.13
```
Comment (fleupold) [inline .travis.yml:16]: How does caching work on github actions?

**Review by fleupold — APPROVED** [review]
Nice!

**Review by josojo — APPROVED** [review]

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (e00E) [reply .github/workflows/pull-request.yaml:12]: should be possible but when would we print it? the job stops when something fails so just printing at the end wouldn't help in that case. if I see an easy to achieve this i'll do it but if not it isn't worth it to make the build script more complicated.

File: .travis.yml (hunk 2 above)
Comment (e00E) [reply .travis.yml:16]: caching looked a little more complicated so i didn't add it to this pr. will measure if sccache speeds up build and look more into caching in another pr.

**Pull request merged**